ios/RNLinksdk.m

```diff
@@ -273,54 +239,433 @@ - (PLKConfiguration*)getLegacyLinkConfiguration:(NSDictionary*)configuration {
-  if ([countryCodes count] > 0) {
-     linkConfiguration.countryCodes = countryCodes;
-  }
-  if ([language length] > 0) {
-     linkConfiguration.language = language;
+      linkConfiguration.accountSubtypes = [RNLinksdk accountSubtypesArrayFromAccountSubtypesDictionary:accountSubtypes];
   }
 
   return linkConfiguration;
 }
 
-@end
+#pragma mark - Bridging
+
++ (PLKEnvironment)environmentFromString:(NSString *)string {
+    if ([string isEqualToString:@"production"]) {
+        return PLKEnvironmentProduction;
+    }
+    
+    if ([string isEqualToString:@"sandbox"]) {
+        return PLKEnvironmentSandbox;
+    }
+    
+    if ([string isEqualToString:@"development"]) {
+        return PLKEnvironmentDevelopment;
+    }
+    
+    // Default to Development
+    NSLog(@"Unexpected environment string value: %@. Expected one of: production, sandbox, or development.", string);
+    return PLKEnvironmentDevelopment;
+}
+
++ (NSArray<NSNumber *> *)productsArrayFromProductsStringArray:(NSArray<NSString *> *)productsStringArray {
+    NSMutableArray<NSNumber *> *results = [NSMutableArray arrayWithCapacity:productsStringArray.count];
+    
+    for (NSString *productString in productsStringArray) {
+        NSNumber *product = [self productFromProductString:productString];
+        if (product) {
+            [results addObject:product];
+        }
+    }
+    
+    return [results copy];
+}
+
++ (NSNumber * __nullable)productFromProductString:(NSString *)productString {
+    NSDictionary *productStringMap = @{
+        @"auth": @(PLKProductAuth),
+        @"identity": @(PLKProductIdentity),
+        @"income": @(PLKProductIncome),
+        @"transactions": @(PLKProductTransactions),
+        @"assets": @(PLKProductAssets),
+        @"liabilities": @(PLKProductLiabilities),
+        @"investments": @(PLKProductInvestments),
+    };
+    return productStringMap[productString.lowercaseString];
+}
+
++ (NSArray<id<PLKAccountSubtype>> *)accountSubtypesArrayFromAccountSubtypesDictionary:(NSDictionary<NSString *, NSArray<NSString *> *> *)accountSubtypesDictionary {
+    __block NSMutableArray<id<PLKAccountSubtype>> *results = [NSMutableArray array];
+    
+    for (NSString *type in accountSubtypesDictionary.allKeys) {
+        NSArray<NSString *> *subtypes = accountSubtypesDictionary[type] ?: @[];
+        
+        for (NSString *subtype in subtypes) {
+            id<PLKAccountSubtype> accountSubtype = [self accountSubtypeFromTypeString:type subtypeString:subtype];
+            if (accountSubtype) {
+                [results addObject:accountSubtype];
+            }
+        }
+    }
+    
+    return [results copy];
+}
+
++ (id<PLKAccountSubtype> __nullable)accountSubtypeFromTypeString:(NSString *)typeString
+                                                   subtypeString:(NSString *)subtypeString {
+    if ([typeString.lowercaseString isEqualToString:@"other"]) {
+        
+    } else if ([typeString.lowercaseString isEqualToString:@"credit"]) {
```

Comment:
ooof this doesn't look fun to write

Reply:
I wrote a script that processed the v2 source code and spat out this if/else-if chain :)